### PR TITLE
created a form to add new Todos

### PR DESCRIPTION
**Pull request by vladgusakovskiy**
[DEMO LINK](https://vladgusakovskiy.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,123 @@
...
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const maxTodoid = todos
+  .map(todo => todo.id)
+  .reduce((a, b) => Math.max(a, b), 0) + 1;
```
Comment (AlinaMatuschak) [inline src/App.tsx:23]: this value still calculates just one time. And each new todo will be with id bigger than 1 that its value (so with the same id)

all new todos with key 16:
Screenshot 2022-09-26 at 22 20 08

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,123 @@
...
+      setIsUserChose(false);
+    }
+
+    if (!title) {
+      setIsTitleChose(false);
+    }
+
+    if (!userId || !title) {
+      return;
+    }
```
Comment (AlinaMatuschak) [inline src/App.tsx:54]: I can create todo where the title has just spaces
Screenshot 2022-09-26 at 22 21 21

File: src/components/TodoList/TodoList.tsx [hunk 3]
```diff
@@ -1 +1,18 @@
...
+};
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
+  return (
+    <section className="TodoList">
+      {todos.map(todo => {
+        return (
+          <TodoInfo key={todo.id} todo={todo} />
+        );
+      })}
```
Comment (AlinaMatuschak) [inline src/components/TodoList/TodoList.tsx:15]:
```suggestion
      {todos.map(todo => (
        <TodoInfo key={todo.id} todo={todo} />
      ))}
```

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,61 +1,122 @@
...
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const todoid = todos.map(todo => todo.id);
+const maxTodoid = Math.max(...todoid) + 1;
```
Comment (AlinaMatuschak) [inline src/App.tsx:22]: Nothing changed ) All new todos have the same id

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]

**Review by AndriiRut — APPROVED** [review]
🔥

**Pull request closed without merging**